packages/frontend/src/hooks/useSavedQuery.ts

```diff
@@ -163,18 +163,43 @@ export const useCreateMutation = () => {
     );
 };
 
-export const useDuplicateMutation = (chartUuid: string) => {
+export const useDuplicateMutation = (
+    chartUuid: string,
+    showRedirectButton: boolean = false,
+) => {
+    const history = useHistory();
     const { projectUuid } = useParams<{ projectUuid: string }>();
     const queryClient = useQueryClient();
     const { showToastSuccess, showToastError } = useApp();
     return useMutation<SavedChart, ApiError, string>(
         () => duplicateSavedQuery(projectUuid, chartUuid),
         {
             mutationKey: ['saved_query_create', projectUuid],
-            onSuccess: async () => {
+            onSuccess: async (data) => {
                 await queryClient.invalidateQueries('spaces');
+
+                if (!showRedirectButton) {
```

Comment:
is there any other place where we use this useDuplicateMutation with showRedirectButton = false ? I can't find anything. I'm wondering if this "automatic redirecting " is the default behaviour we want

Reply:
@rephus We use it with `false` in the explorer page, if you see the comments in [the issue ] from @PriPatel 
 (https://github.com/lightdash/lightdash/issues/1767). the behaviour we want is:

Scenario 1:
User duplicates **_charts_** from saved **_charts list_** -> Success message will display **with link** to the newly created **_chart_**

Scenario 2:
User duplicates **_dashboards_** from saved **_dashboards list_** -> Success message will display **with link** to the newly created **_dashboard_**

Scenario 3:
User duplicates **_chart_** from the **_explorer page_** -> user will be automatically **redirected** to the newly created **_chart_**